pom.xml

```diff
@@ -381,6 +386,17 @@
     </profiles>
 
     <dependencies>
+        <!--
+        This dependency makes compilation on non-GraalVM versions possible.
+        The dependency should however never actually be required by end-users because if they require
+        GraalVM features, they will be provided by the JDK they are using.
+        -->
+        <dependency>
+            <groupId>org.graalvm.sdk</groupId>
+            <artifactId>graal-sdk</artifactId>
+            <version>${graalvm.version}</version>
+            <scope>provided</scope>
+        </dependency>
```

Comment:
Sidenote on this, provided scope works perfectly fine as discussed earlier. Seems like my understanding of the provided scope was wrong.

Reply:
Thanks for looking into this (: I added a comment in the other thread in case others reference it, and perhaps there's something we're missing here.